prometheus/prometheus.go

```diff
@@ -252,11 +252,16 @@ var forbiddenChars = regexp.MustCompile("[ .=\\-/]")
 
 func flattenKey(parts []string, labels []metrics.Label) (string, string) {
 	key := strings.Join(parts, "_")
-	key = forbiddenChars.ReplaceAllString(key, "_")
+
+	// Only run the regex replace if the key contains forbidden characters to
+	// void overhead on well-formed keys.
```

Comment:
```suggestion
	// avoid overhead on well-formed keys.
```

I wonder if it's worth a link to this PR to show this wasn't a premature optimization and actually helps?